lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -88,22 +88,37 @@ module.exports = {
 
   // Gives the least recent last modify date across all the functions in the service.
   async getFunctionsEarliestLastModifiedDate() {
+    let fetchError;
     const getFunctionResults = this.serverless.service.getAllFunctions().map((funName) => {
       const functionObj = this.serverless.service.getFunction(funName);
       return this.provider
         .request('Lambda', 'getFunction', {
           FunctionName: functionObj.name,
         })
         .then((res) => new Date(res.Configuration.LastModified))
-        .catch(() => new Date(0)); // Function is missing, needs to be deployed
+        .catch((err) => {
+          fetchError = err;
+          return new Date(0);
+        }); // Function is missing, needs to be deployed
     });
 
-    return BbPromise.all(getFunctionResults).then((results) =>
-      results.reduce((currentMin, date) => {
+    return BbPromise.all(getFunctionResults).then((results) => {
+      if (fetchError && fetchError.providerError && fetchError.providerError.statusCode === 403) {
+        this.serverless.cli.log(
+          [
+            'WARNING: Iam policy misconfiguration: unable to get Lambda functions info.',
+            'Deployment will not be skipped even if service files did not change. ',
+            'Forcing deployment...',
```

Comment:
I think we can remove `Forcing deployment...` part in the message, as it's already understood what will happen from the line above.